lib/util/print.js

```diff
@@ -91,4 +96,25 @@ function printTreeTraversal(node) {
   }
 }
 
+// Need to add parens around any [+, -] operation
+// next to [/, *, ^] operation or parenthesis node
+// Examples:
+// { "/" [{ "+" ["x", "2"] }, "2"] } -> (x + 2) / 2.
+// { "+" [{ "+" ["x", "x"] }, { "+" ["2", "2"] }] } -> (x + x) + (2 + 2)
+// Check #120, #126 issues for more details.
+function needToAddParens(node, parentNode) {
+  if (!parentNode ||
+      !Node.Type.isOperator(parentNode) ||
+      '*/^'.indexOf(node.op) !== -1) {
+    return false;
+  }
+
+  if ('+-'.indexOf(parentNode.op) === -1 ||
+      parentNode.args.some(Node.Type.isParenthesis)) {
```

Comment:
No, current node has to be Operation type, so 2 alone cannot be wrapped. But it does not matter, if 3 + 4 + (4 - 5) is preferable over (3 + 4) + (4 - 5). I will remove it nonetheless.